### PR TITLE
Adds slack notifcation

### LINKED ISSUE #122: Add Slack Notifications/Alerts
It would be nice if you can configure the sanity-runner to (optionally) send slack notifications when a test fails. Using `Pragmas` in the top level comment we could specify a custom channel per test for it to alert to, or it could fall back to top level config default channel.

Information that can be included in alert:
- ENV/all APP_ENVs
- test name
- list description of test (set in `Pragmas`) 
- screenshot of failure

### PR DESCRIPTION
**Pull request by mhaber-tophat**
closes #122 

Adds support for alerting to slack. 

requires `ALERT=true` to be set as a variables. This allows the user the option to turn off notification/not send to slack if they testing a local branch.

requires `sanity_runner/slack_api_token` to exist within aws secretmanager

Picks up Slack channel + test description via pragams in top level comment of a test. 

EX.
```
/** 
 * @Owner Fake Team Name
 * @Slack #itsachannelname
 * @Description 
 *  - test
**/

import { setDefaultOptions } from 'expect-puppeteer'
const constants = require('../constants/constants');

describe('Baseline', async () => {
    setDefaultOptions({ timeout: 6000 })

    test('Tophat responds successfully', async () => {
        await expect("asdasdas").toEqual("123123")
    },900000)
})
``` 
the above test will result in an alert like the below
Screen Shot 2019-09-09 at 3 23 10 PM

### CODE REVIEW

File: service/src/alertOnResult.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            console.log('Slack Token was not retrieved')
+            return 1
+        }
+        const slack = new WebClient(slackToken.slack_api_token)
+        const slackChannel = testMetaData.Slack
+
+        const testResults = results.testResults[0]
+
+        //obtain all screen shots
+        const screenShotAttachment = []
```
Comment (chrono) [inline service/src/alertOnResult.js]:
```suggestion
        const screenShotAttachments = []
```

File: service/src/alertOnResult.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        for (const screenshotTitle of Object.keys(results.screenshots)) {
+            screenShotAttachment.push({
```
Comment (chrono) [inline service/src/alertOnResult.js]:
```suggestion
            screenShotAttachments.push({
```

File: service/src/alertOnResult.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                color: '#D40E0D',
+            })
+        }
+
+        // Send Slack message and format into thread
+        const parentMessage = `Sanity... \`${test}\` has Failed!`
+        const resParent = await slack.chat.postMessage({
+            channel: slackChannel,
+            text: parentMessage,
+            attachments: screenShotAttachment,
```
Comment (chrono) [inline service/src/alertOnResult.js]:
```suggestion
            attachments: screenShotAttachments,
```

File: service/src/alertOnResult.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        })
+
+        const descriptionMessage = testMetaData.Description
+        await slack.chat.postMessage({
+            channel: slackChannel,
+            thread_ts: resParent.ts,
+            attachments: [
+                {
+                    title: 'Manual Steps for Sanity',
+                    text: descriptionMessage.replace(/-/gi, '\n- '),
```
Comment (chrono) [inline service/src/alertOnResult.js]: any dash anywhere get's turned into a list?

Comment (mhaber-tophat) [reply service/src/alertOnResult.js]: added comment in code on why

**Pull request merged**